### PR TITLE
Return actual file size file info in RemoteFileList

### PR DESCRIPTION
**Pull request by iamrz1**
Upload files to dStorage using io reader (from local or remote locations). It helps to compare file sizes and make an informed decision.

### CODE REVIEW

File: zboxcore/sdk/sync.go [hunk 1]
```diff
@@ -29,9 +29,10 @@ const (
 )
 
 type fileInfo struct {
```
Comment (moldis) [inline zboxcore/sdk/sync.go:31]: Actual Size is already added to fileRef
https://github.com/0chain/gosdk/commit/a7eeb212ef107e889f80bdf90c274b9c8f41019f

**Review by moldis — CHANGES_REQUESTED** [review]
Please, check my comments, not sure if this branch needed

**Comment by iamrz1** [comment]
Okay, that will do.

**Comment by iamrz1** [comment]
I guess there are a few structs that returns fileInfo, this one is in sync.go. this struct is used in GetAllocation(). So I think we need this pr.

File: core/zcncrypto/bls0chain_test.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func BenchmarkBLSSign(b *testing.B) {
 func TestRecoveryKeys(t *testing.T) {
 
 	sigScheme := NewSignatureScheme("bls0chain")
-    TestSignatureScheme(t)
+	TestSignatureScheme(t)
```
Comment (Sriep) [inline core/zcncrypto/bls0chain_test.go:61]: No meaningful changes here, revert changes to this file.

File: zboxcore/sdk/allocation_test.go [hunk 3]
```diff
@@ -31,7 +31,7 @@ func TestGetMinMaxWriteReadSuccess(t *testing.T) {
 		maxW, maxR, err := ssc.GetMaxWriteRead()
 		require.NoError(t, err)
 		require.Equal(t, 0.01, maxW)
-		require.Equal(t,0.01, maxR)
+		require.Equal(t, 0.01, maxR)
```
Comment (Sriep) [inline zboxcore/sdk/allocation_test.go:34]: No meaningful changes here, revert changes to this file.

File: core/zcncrypto/zcncrypto.go [hunk 4]
```diff
@@ -46,6 +46,7 @@ type SignatureScheme interface {
 	// Combine signature for schemes BLS
 	Add(signature, msg string) (string, error)
 }
+
```
Comment (Sriep) [inline core/zcncrypto/zcncrypto.go:49]: No meaningful changes here, revert changes to this file.

**Review by Sriep — CHANGES_REQUESTED** [review]
Three files with no real changes, need to be reverted. Only `sync.go` belongs in this PR.

Adding a field looks harmless, but I can't find anywhere it is used. Is there a link to it being used? Or is there a follow-up PR that will make use of the field?

**Comment by iamrz1** [comment]
> Three files with no real changes, need to be reverted. Only `sync.go` belongs in this PR.
> 
> Adding a field looks harmless, but I can't find anywhere it is used. Is there a link to it being used? Or is there a follow-up PR that will make use of the field?

I removed the other fmt related changes.
There's a PR in zbox-cli (https://github.com/0chain/zboxcli/pull/21) that uses actual file size to compare existing file (in dStorage) with source file to determine if that upload was complete.

**Review by Sriep — APPROVED** [review]

**Pull request merged**